test/micromamba/test_shell.py

```diff
@@ -145,68 +157,65 @@ def custom_shell(shell):
-            )
-        ):
-            pytest.skip("Incompatible shell/OS")
+        skip_if_shell_incompat(shell_type)
 
-        if not root and env_exists:
+        if env_exists:
+            # Create the environment for this test, so that it exists
             create("-n", TestShell.env_name, "-q", "--offline", no_dry_run=True)
+        else:
+            shutil.rmtree(TestShell.root_prefix)
+
+        if root:
+            p = TestShell.root_prefix
+            n = "base"
+        else:
+            p = TestShell.prefix
+            n = TestShell.env_name
 
         if prefix_type == "prefix":
             if expanded_home:
-                cmd = ("activate", "-s", shell_type, "-p", TestShell.prefix)
+                cmd = ("activate", "-s", shell_type, "-p", p)
             else:
                 cmd = (
                     "activate",
                     "-s",
                     shell_type,
                     "-p",
-                    TestShell.prefix.replace(os.path.expanduser("~"), "~"),
+                    p.replace(os.path.expanduser("~"), "~"),
                 )
         else:
-            cmd = ("activate", "-s", shell_type, "-p", TestShell.env_name)
+            cmd = ("activate", "-s", shell_type, "-p", p)
 
-        res = shell(*cmd)
+        if env_exists:
+            res = shell(*cmd)
+        else:
+            with pytest.raises(subprocess.CalledProcessError):
+                shell(*cmd)
+            return
 
         # TODO: improve this test
         assert res
 
         if shell_type == "bash":
-            assert f"export CONDA_PREFIX='{TestShell.prefix}'" in res
-            assert f"export CONDA_DEFAULT_ENV='{TestShell.env_name}'" in res
-            assert f"export CONDA_PROMPT_MODIFIER='({TestShell.env_name}) '" in res
+            assert f"export CONDA_PREFIX='{p}'" in res
+            assert f"export CONDA_DEFAULT_ENV='{n}'" in res
+            assert f"export CONDA_PROMPT_MODIFIER='({n}) '" in res
 
     @pytest.mark.parametrize("shell_type", ["bash", "powershell", "cmd.exe"])
     @pytest.mark.parametrize("prefix_selector", [None, "prefix"])
     def test_init(self, shell_type, prefix_selector):
-        if (
-            (platform.system() == "Linux" and shell_type != "bash")
-            or (
-                platform.system() == "Windows"
-                and shell_type not in ("cmd.exe", "powershell")
-            )
-            or (platform.system() == "Darwin" and shell_type not in ("zsh", "bash"))
-        ):
-            pytest.skip("Incompatible shell/OS")
+        skip_if_shell_incompat(shell_type)
 
         if prefix_selector:
             shell("-y", "init", "-s", shell_type, "-p", TestShell.root_prefix)
+            assert (
+                Path(os.path.join(TestShell.root_prefix, "condabin")).is_dir()
+                or Path(
+                    os.path.join(TestShell.root_prefix, "etc", "profile.d")
+                ).is_dir()
```

Comment:
```suggestion
                Path(os.path.join(TestShell.root_prefix, "condabin")).is_dir()
                and Path(
                    os.path.join(TestShell.root_prefix, "etc", "profile.d")
                ).is_dir()
```

Reply:
depending on the shell, we expect file at one place OR the other